pyproject.toml

```
@@ -39,9 +39,11 @@ dependencies = [
 
 [project.optional-dependencies]
 test = [
+    "black~=23.7.0",
     "httpx~=0.23.0",
     "pytest~=7.1.2",
     "pytest-asyncio~=0.18.3",
+    "ruff~=0.0.287",
     "websockets~=10.3"
 ]
```

I would propose to have it like the following:

```suggestion
[project.optional-dependencies]
test = [
    "httpx~=0.23.0",
    "pytest~=7.1.2",
    "pytest-asyncio~=0.18.3",
    "websockets~=10.3"
]
lint = [
    "black~=23.7.0",
    "ruff~=0.0.287",
    "mypy~=1.5.1"
]
dev = ["granian[test,lint]"]
```

Having the `dev` is optional, of course. It’s just a convenience.